src/test/java/org/prebid/server/bidder/invibes/InvibesBidderTest.java

```diff
@@ -0,0 +1,242 @@
+        // given
+        final BidRequest bidRequest = givenBidRequest(identity(), ExtImpInvibes
+                .of("12", 15, InvibesDebug.of("test", true))).toBuilder()
+                .site(Site.builder().page("www.awesome-page.com").build()).build();
+
+        // when
+        final Result<List<HttpRequest<InvibesBidRequest>>> result = invibesBidder.makeHttpRequests(bidRequest);
+
+        // then
+        final InvibesBidRequest expectedRequest = InvibesBidRequest.builder()
+                .bidParamsJson("{\"PlacementIds\":[\"12\"],\"BidVersion\":\"4\","
+                        + "\"Properties\":{\"12\":{\"Formats\":[{\"w\":15,\"h\":12}]}}}")
+                .isTestBid(Boolean.TRUE)
+                .location("www.awesome-page.com")
+                .gdpr(Boolean.TRUE)
+                .gdprConsent(StringUtils.EMPTY)
+                .invibBVLog(Boolean.TRUE)
+                .videoAdDebug(Boolean.TRUE)
+                .lid(StringUtils.EMPTY)
+                .bvid("test")
+                .build();
+
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1).extracting(HttpRequest::getPayload).containsOnly(expectedRequest);
+    }
+
+    @Test
+    public void makeBidsShouldReturnErrorWhenResponseBodyCouldNotBeParsed() {
+        // given
+        final HttpCall<InvibesBidRequest> httpCall = givenHttpCall(null, "invalid");
+
+        // when
+        final Result<List<BidderBid>> result = invibesBidder.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1);
+        assertThat(result.getErrors().get(0).getMessage()).startsWith("Failed to decode: Unrecognized token");
+        assertThat(result.getErrors().get(0).getType()).isEqualTo(BidderError.Type.bad_server_response);
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeBidsShouldReturnEmptyListWhenBidResponseIsNull() throws JsonProcessingException {
+        // given
+        final HttpCall<InvibesBidRequest> httpCall = givenHttpCall(null,
+                mapper.writeValueAsString(null));
+
+        // when
+        final Result<List<BidderBid>> result = invibesBidder.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeBidsShouldReturnBannerBid() throws JsonProcessingException {
+        // given
+        final HttpCall<InvibesBidRequest> httpCall = givenHttpCall(
+                InvibesBidRequest.builder()
+                        .build(),
+                mapper.writeValueAsString(
+                        givenBidResponse(bidBuilder -> bidBuilder.impid("123"))));
+
+        // when
+        final Result<List<BidderBid>> result = invibesBidder.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue())
```

Comment:
.extracting(BidderBid::getBid)

Reply:
idea of this test to check is it banner bid or not, extracting will not give possibility to check like this 
BidderBid.of(Bid.builder().impid("123").build(), banner, "EUR")